### PR TITLE
Add the two probabilistic strategies

### PR DESCRIPTION
**Pull request by norhh**
currently implemented the first 2 strategies mentioned in #539 .

### CODE REVIEW

File: mythril/laser/ethereum/strategy/basic.py [hunk 1]
```diff
@@ -1,54 +1,80 @@
...
+    from bisect import bisect
+
+    def choices(population, weights=None):
+        if weights is None:
+            return [population[int(random() * len(population))]]
+        cum_weights = accumulate(weights)
+        return [population[bisect(cum_weights, random()*cum_weights[-1], 0, len(population)-1)]]
+
+
+class BasicStrategy(ABC):
```
Comment (JoranHonig) [inline mythril/laser/ethereum/strategy/basic.py]: I'd name this SearchStrategy.
Also can you put this in __init__ ? I think that is a better place for this class

File: mythril/laser/ethereum/strategy/basic.py [hunk 2]
```diff
@@ -1,54 +1,80 @@
...
 
+try:
+    from random import choices
+except ImportError:
+
+    from itertools import accumulate
+    from random import random
+    from bisect import bisect
+
+    def choices(population, weights=None):
```
Comment (JoranHonig) [inline mythril/laser/ethereum/strategy/basic.py]: pydoc?

File: mythril/laser/ethereum/strategy/basic.py (hunk 1 above)
Comment (norhh) [reply mythril/laser/ethereum/strategy/basic.py]: Let’s add the ‘basic’ term as a prefix as not all search strategies can be emulated with this class.how about calling it something like ‘BasicSearchStrategy’.

File: mythril/laser/ethereum/strategy/basic.py (hunk 2 above)
Comment (norhh) [reply mythril/laser/ethereum/strategy/basic.py]: Sorry, I didn't get what exactly to be done, if it is comments about the function then I have just added it now.

**Comment by norhh** [comment]
I think it might be good to construct a graph now for the 3rd strategy as it might be possible to try out a lot more strategies easily if we have a graph for the worklist.I will put that PR after this gets reviewed as it might be good to have smaller PRs for review.

File: mythril/analysis/symbolic.py [hunk 3]
```diff
@@ -21,6 +22,10 @@ def __init__(self, contract, address, strategy, dynloader=None, max_depth=22,
             s_strategy = DepthFirstSearchStrategy
```
Comment (JoranHonig) [inline mythril/analysis/symbolic.py]: What do you think about the following:
Modules in laser.strategy all define some metadata and are dynamically loaded vs doing it hardcoded here

Comment (JoranHonig) [reply mythril/analysis/symbolic.py]: This should be a separate pr ofc

Comment (norhh) [reply mythril/analysis/symbolic.py]: :+1:

**Pull request merged**